h/schemas/forms/admin/organization.py

```diff
@@ -2,20 +2,39 @@
 
 import colander
 from deform.widget import TextAreaWidget, TextInputWidget
+from xml.etree import ElementTree
 
 import h.i18n
+from h.models.organization import (
+    ORGANIZATION_NAME_MIN_CHARS,
+    ORGANIZATION_NAME_MAX_CHARS,
+)
 from h import validators
-from h.models.organization import Organization, ORGANIZATION_NAME_MIN_CHARS, ORGANIZATION_NAME_MAX_CHARS
 from h.schemas.base import CSRFSchema
 
 _ = h.i18n.TranslationString
 
 
+ORGANIZATION_LOGO_MAX_CHARS = 100000
+
+
+def _strip_xmlns(tag):
+    if '}' in tag:
+        return tag.split('}', 1)[1]
+    return tag
+
+
 def validate_logo(node, value):
+    if len(value) > ORGANIZATION_LOGO_MAX_CHARS:
+        raise colander.Invalid(node, _(
+            "Logo is larger than {:,d} characters".format(ORGANIZATION_LOGO_MAX_CHARS)))
     try:
-        Organization(logo=value)
-    except Exception:
-        raise colander.Invalid(node, _('Logo is not valid SVG'))
+        root = ElementTree.fromstring(value)
+    except ElementTree.ParseError:
+        raise colander.Invalid(node, _('Logo is not a valid SVG'))
```

Comment:
This message has lost a useful piece of information from before which is that the XML couldn't be parsed, as opposed to other reasons why the content might not be a valid SVG.

Reply:
done